x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,172 @@
+# Timeline Batching Functional Design ADR
+
+
+
+This document details the design of how Timeline Batching is supposed to work. The purpose of this document is to have a source of truth where reviewers of any changes can refer back and decide if the change is in line with the design or not.
+
+This document is only applicable for Timeline datagrid.
+
+## What is a batch
+Batch ( also known as Sample ) in Unified Data table is the collection of max number of documents that are fetched in a single request. When a user queries a timeline for the first time, Batch 0 is fetched and then the user may have the option to fetch further batches on-demand.
+
+
+### Batch v/s Data Grid Pagination
+The concept of batch is different from the Datagrid Pagination. Datagrid Pagination is completely client driven and datagrid may decide how it wants to paginate a particular batch.
+
+For example, let’s say we fetch a batch of 500 documents, now Datagrid may decide to display these 500 documents following configurations 
+10 pages with each page consisting of 50 documents  
+20 pages with each page consisting of 25 documents
+5 pages with each page consisting of 100 documents.
+
+Users will be able to fetch a new batch of documents when they reach the last page.
+
+### Batch Parameters
+Below are the parameters that affect a batch. Any changes to these parameters should trigger a re-fetch of all existing batches or the initial batch depending on the situation.
+
+- **dataViewId**: Currently selected dataView Id in Timeline.
+- **Timerange**: 
+  - **startDate**
+  - **endDate**
+  - **timeRangeKind**
+- **eqlOptions**
+- **indexNames**: List of indices in Timeline.
+- **id**: Timeline Id
+- **fields**:  List of requested fields
+- **filterQuery**:  combined query consisting: 
+  * **KQL/Lucene Query**
+  * **Filters**
+  * **Data Providers**
+- **runtimeMappings**: List of runtime mappings that need to be fetched
+- **language**: KQL | Lucene
+- **limit**: Sample Size or Batch Size
```

Comment:
We may be able to remove limit as it should act on the existing batches? If more is requested, then you would just request new batches right?

Reply:
>We may be able to remove limit as it should act on the existing batches?

Sorry I do not understand this statement.


>If more is requested, then you would just request new batches right?

Exactly. If you check line 163 ( section batch size ), we fetch initial batch.

Reply:
> Sorry I do not understand this statement.

Sorry, yea... should've been clearer! Do all existing batches need to be re-fetched if limit changes?

Reply:
No.. we only fetch first batch again. It will be like `Refetch` button has been pushed or user is starting again.

Reply:
You can ignore me, I guess it's just a sane default to do that. I was also wondering if we refetch all batches if the user has absolute time ranges set, and it looks like other parts of the application do that as well. 👍🏾  Thanks